project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -355,6 +421,45 @@ record ProjectCreationRequest(ProjectDesign design, ProjectContacts contacts,
 
   }
 
+  /**
+   * A service request to create one or more new samples for a project.
+   *
+   * @param projectId the project ID of the project the samples shall be created for
+   * @param requests  a collection of {@link SampleRegistrationRequest} items
+   * @since 1.10.0
+   */
+  record SampleCreationRequest(String projectId, Collection<SampleRegistrationRequest> requests) {
+    public SampleCreationRequest(String projectId, Collection<SampleRegistrationRequest> requests) {
+      this.projectId = projectId;
+      this.requests = List.copyOf(requests);
```

Comment:
You need to overwrite the getter for the `requests` as well to ensure they are not altered by accessing methods.